### PR TITLE
Refactor logprobs

### PR DESCRIPTION
**Pull request by faresobeid**
Makes recomputing logprobs only for logging purposes, and renames logprobs

### CODE REVIEW

**Review by mikasenghaas — APPROVED** [review]
lgtm!

**Comment by mikasenghaas** [comment]
note to self: we're delaying this one because of areal (#980)

File: src/prime_rl/trainer/rl/loss.py [hunk 1]
```diff
@@ -82,43 +82,51 @@ def compute_loss(
...
+    for trainer_logprobs, inference_logprobs, advantages, loss_mask in zip(
+        trainer_logprobs, inference_logprobs, advantages, loss_mask
+    ):
+        total_tokens = torch.clamp_min(loss_mask.sum(), 1)
+
+        log_importance_ratio = trainer_logprobs - inference_logprobs
+
+        # Compute trainer-inference mismatch KL
+        mismatch_kl = (torch.exp(log_importance_ratio) - log_importance_ratio - 1)
+        mismatch_kl = mismatch_kl[loss_mask].sum().unsqueeze(0) / total_tokens
```
Comment (samsja) [inline src/prime_rl/trainer/rl/loss.py:97]:
```suggestion
with torch.no_grad()
        log_importance_ratio = trainer_logprobs - inference_logprobs

        # Compute trainer-inference mismatch KL
        mismatch_kl = (torch.exp(log_importance_ratio) - log_importance_ratio - 1)
        mismatch_kl = mismatch_kl[loss_mask].sum().unsqueeze(0) / total_tokens
```

**Review by samsja — COMMENTED** [review]
can we delete the following function

wake_up_model_from_cpu
copy_model_to_cpu
offload_model_to_cpu

OffloadedTensor
